src/components/PersonalDataForm.jsx

```diff
@@ -0,0 +1,94 @@
+import React from 'react'
+import { useForm } from 'react-hook-form'
+import { useState } from 'react'
+import { SummaryDataForm } from './SummaryDataForm'
+import { zodResolver } from '@hookform/resolvers/zod'
+import { schemaForm } from '../schemas/schemaForm'
+import { Error, Form, Input, Select, Option, Header, Label, ChkContainer, Button } from '../ui'
+import { ExtraExperience } from './ExtraExperience'
+
+// Extracted data
+const defaultValues = { experiences: [] }
+const learningModes = schemaForm._def.schema.shape.learningMode.options // Extract the enum values
+const technologyOptions = schemaForm._def.schema.shape.technologies.element.options // Extract the array values
+const expTechnologies = schemaForm._def.schema.shape.experiences.element.shape.technology.options // Extract the enum values for exp tech
+const expLevels = schemaForm._def.schema.shape.experiences.element.shape.level.options // Extract the enum values for exp lvl
+
+export const PersonalDataForm = () => {
+   const {
+      register,
+      handleSubmit,
+      watch,
+      control,
+      formState: { errors },
+   } = useForm({
+      resolver: zodResolver(schemaForm),
+      defaultValues: defaultValues,
+   })
+
+   const [userData, setUserData] = useState(null)
+   const watchExperience = watch('experience', false)
+
+   const onSubmit = data => {
+      setUserData(data)
+   }
+
+   return userData ? (
+      <SummaryDataForm userData={userData} />
+   ) : (
+      <>
+         <Header data-type="h1">Formularz zgłoszeniowy na kurs programowania</Header>
+         <Form id="personalDataForm" onSubmit={handleSubmit(onSubmit)}>
+            <Header data-type="h2"> Dane osobowe</Header>
+            <Input type="text" id="firstName" name="firstName" placeholder="Imię" {...register('firstName')} />
```

Comment:
The strange text formatting indicates an issue with the Prettier plugin's functioning.